pkg/reconciler/instances/istio/reset/data/gatherer.go

```diff
@@ -59,13 +59,26 @@ func (i *DefaultGatherer) GetPodsWithDifferentImage(inputPodsList v1.PodList, im
 		for _, container := range pod.Spec.Containers {
 			containsPrefix := strings.Contains(container.Image, image.Prefix)
 			hasSuffix := strings.HasSuffix(container.Image, image.Version)
-			isTerminating := pod.Status.Phase == "Terminating"
+			ready := isPodReady(pod)
 
-			if containsPrefix && !hasSuffix && !isTerminating {
+			if containsPrefix && !hasSuffix && ready {
 				outputPodsList.Items = append(outputPodsList.Items, *pod.DeepCopy())
 			}
 		}
 	}
 
 	return
 }
+
+func isPodReady(pod v1.Pod) bool {
+	if pod.Status.Phase != v1.PodRunning {
+		return false
+	}
+	for _, condition := range pod.Status.Conditions {
+		if condition.Status != v1.ConditionTrue {
+			return false
+		}
+	}
+	//deletion timestamp determines whether pod is terminating or running (nil == running)
```

Comment:
Instead of this comment I would rather see a function with a proper name and docs:
```go
func isPodRunning(pod v1.Pod) bool {
  return pod.ObjectMeta.DeletionTimestamp == nil
}
```